services/nats/pkg/server/nats/nats.go

```diff
@@ -44,6 +47,7 @@ func (n *NATSServer) ListenAndServe() (err error) {
 	return nil
 }
 
+// Shutdown stops the NATSServer gracefully
 func (n *NATSServer) Shutdown() {
 	n.server.Shutdown()
```

Comment:
I think you should add 

```
 n.server.WaitForShutdown()
```

here. (Similar to what the signal handler in nats does)